src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -51,16 +53,46 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
             _ackHandler = new AckHandler();
 
             var writer = new LoggerTextWriter(logger);
-            _logger.ConnectingToEndpoints(string.Join(", ", options.Value.Options.EndPoints.Select(e => EndPointCollection.ToString(e))));
+            _logger.ConnectingToEndpoints(options.Value.Options.EndPoints);
             _redisServerConnection = _options.Connect(writer);
+
+            _redisServerConnection.ConnectionRestored += (_, e) =>
+            {
+                lock(_redisConnectionEventLock)
+                {
+                    if (_connectionState == State.Connected)
```

Comment:
Why do we need it? Do we get an event per Subscription?

Reply:
https://github.com/SignalR/SignalR/pull/3849/files#diff-193a338ff4a86e8ec19aa6cfa2954768L242

Reply:
In old SignalR we also have an Interactive connection in addition to pub sub and this was the [source of duplicate](https://github.com/SignalR/SignalR/issues/3729#issuecomment-261760333) ConnectionRestored/ConnectionClosed events.

Reply:
Also, because you only log this you could log all events and just provide the connection type or anything else that can differentiate. Might be even better/truer

Reply:
I don't think it would be useful to see "Connection restored for Interactive connection" and "Connection restored for Subscription connection"?

Reply:
OK. (Maybe it is moot point anyways because AFAIR we don't have anyInteractive connection)